extensions/amp-analytics/0.1/scroll-manager.js

```diff
@@ -76,16 +91,27 @@ export class ScrollManager {
   addScrollHandler(handler) {
     // Trigger an event to fire events that might have already happened.
     const size = this.viewport_.getSize();
-    /** {./scroll-manager.ScrollEventDef} */
-    const scrollEvent = {
-      top: this.viewport_.getScrollTop(),
-      left: this.viewport_.getScrollLeft(),
-      width: size.width,
-      height: size.height,
-      scrollWidth: this.viewport_.getScrollWidth(),
-      scrollHeight: this.viewport_.getScrollHeight(),
-    };
-    handler(scrollEvent);
+
+    this.getInitRootElementRect_().then(layoutRect => {
+      const {
+        top: scrollTop,
+        left: scrollLeft,
+        width: scrollWidth,
+        height: scrollHeight,
+      } = layoutRect;
+
+      /** {./scroll-manager.ScrollEventDef} */
+      const scrollEvent = {
+        top: this.viewport_.getScrollTop() - scrollTop,
```

Comment:
Most ideally I'd still want viewport implementation to offset the scrollTop. But this is works. Please kindly add some comment.